frontend/src/components/authentication/ForgotPassword.tsx

```diff
@@ -0,0 +1,90 @@
+import React, {useState, useCallback} from 'react'
+import {useDispatch} from 'react-redux'
+import {CssBaseline, TextField} from '@material-ui/core'
+import {Button, Avatar, Typography, FormControlLabel, Checkbox, Link, Paper, Grid }  from '@material-ui/core'
+import LockOutlinedIcon from '@material-ui/icons/LockOutlined'
+
+import * as gridSize from '../../constants/styles.values'
+import {SIX as SHADOW_DEPTH_SIX} from '../../constants/styles.values'
+import { recover } from '../../actions/authActions'
+
+import {useStyles} from './ForgotPassword.styles'
+
+export default function ForgotPasswordSide(): React.ReactElement {
+
+    const [email, setEmail] = useState('')
+    const onChangeEmail = useCallback((e) => {
+      setEmail(e.target.value)
+    }, [setEmail])
+  
+    const dispatch = useDispatch()
+  
+    const recoverPassword = useCallback((e) => {
+      e.preventDefault()
+      // eslint-disable-next-line
+      // @ts-ignore
+      dispatch(recover({email}))
+    }, [dispatch, email])
+  
+    const classes = useStyles()
+  
+    return (
+      <Grid className={classes.root} component="main" container>
+        <CssBaseline />    
+        <Grid component={Paper}
+              elevation={SHADOW_DEPTH_SIX}
+              item md={gridSize.FIVE}
+              sm={gridSize.EIGHT}
+              square
+              xs={gridSize.TWELVE}
+              >
+        <div className={classes.paper}>
+          <Avatar className={classes.avatar}>
+            <LockOutlinedIcon />
+          </Avatar>
+          <Typography component="h1" variant="h5">
+            Recover password
+          </Typography>
+          <form className={classes.form} noValidate>
+            <Grid container spacing={2}>
+              <Grid item xs={12}>
+                Enter your e-mail to recover your password.
+              </Grid>
+              <Grid item xs={12}>
+                <TextField
+                  autoComplete="email"
+                  fullWidth
+                  id="email"
+                  label="Email Address"
+                  name="email"
+                  onChange={onChangeEmail}
```

Comment:
Please add validation that its email before its sent to backend, maybe use Formik and Yup for this (examples are in nickname or password changing files) - it will make code qty lesser than with use of useState.

On backend please change settings for error display on response - 
`You're seeing this error because you have DEBUG = True in your
Django settings file. Change that to False, and Django will
display a standard page generated by the handler for this status code.`
This way error message is too long and cant be handled via errorHandler.

Reply:
I fixed reported bugs and implemented all the suggested changes. You may now pull and check them out. As to email validation, I used a regex. If the submitted email conforms to the regular expression, the email is then sent to backend. If not, an error toast appears saying "Invalid email format"